src/context/VernaContextProvider.tsx

```diff
@@ -182,8 +159,9 @@ function VernaContextProvider({
 
   function handleSubmit<FormData = unknown>(callback?: (formData: unknown) => void) {
     return (event: ISubmitEvent<FormData>) => {
-      const newFormData = { ...formData };
+      let newFormData = { ...formData };
       if (selector) newFormData[selector] = event.formData;
+      else newFormData = event.formData;
```

Comment:
Shorter and use of a `const`.
```suggestion
      const newFormData = selector ? { ...formData, [selector]: event.formData } : event.formData;
```

Reply:
Nice one !